hw/ip/ascon/data/ascon.hjson

```diff
@@ -0,0 +1,696 @@
+        Any write operation to this register will clear the status tracking required for automatic mode (See CTRL_AUS_SHADOWED.manual_start_trigger).
+        A write to the Control Register is considered the start of a new message.
+        Hence, software needs to provide a new Nonce and input data afterwards.
+      '''
+      swaccess: "rw",
+      hwaccess: "hro",
+      hwext:    "false"
+      hwqe:     "true"
+      shadowed: "true"
+      fields: [
+        { bits: "1:0",
+          name: "OPERATION",
+          desc:  '''
+            Specifies which operation ascon should perform. There are
+            Enc, Dec, Hash (not implemented yet)
+          '''
+          enum: [
+            { value: "1",
+              name: "ASCON_ENC",
+              desc: '''
+                2'b01: Encryption.
+                Invalid input values (2'b00) mapped to ASCON_ENC.
+              '''
+            },
+            { value: "2",
+              name: "ASCON_DEC",
+              desc: '''
+                2'b10: Decryption.
+              '''
+            },
+            { value: "3",
+              name: "ASCON_HASH",
+              desc: '''
+                '2b11: Hash.
+              '''
+            },
+          ]
+        }
+        { bits: "2",
+          name: "SIDELOAD_KEY",
+          desc:  '''
+            Controls whether the Ascon unit uses the key provided by the key manager via key sideload interface (1)
+            or the key provided by software via Initial Key Registers KEY_SHARE1 and KEY_SHARE_0 (0).
+          '''
+        }
+        { bits: "3",
+          name: "MASKED_AD_INPUT",
+          desc:  '''
+            Controls whether the associated data input is provided in shares (1) or not (0).
+            If the input is provided in shares all registers of both shares must be written for each input block.
+            If set to 0, share 1 must be set to all zeros at the beginning. It does not need to be rewritten for each block.
+            Once all registers of share 0 have been written, Ascon starts to process the data depending on the CTRL_AUS_SHADOWED_manual_start_trigger.
+          '''
+        }
+        { bits: "4",
+          name: "MASKED_MSG_INPUT",
+          desc:  '''
+            Controls whether the message input is provided in shares (1) or not (0).
+            If the input is provided in shares all registers of both shares must be written for each input block.
+            If set to 0, share 1 must be set to all zeros at the beginning. It does not need to be rewritten for each block.
+            Once all registers of share 0 have been written, Ascon starts to process the data depending on the CTRL_AUS_SHADOWED_manual_start_trigger.
+          '''
+        }
+        { bits: "5",
+          name: "NO_MSG",
+          desc:  '''
+            There is no (1) message (plaintext/ciphertext) to be processed. There is (0) a message.
```

Comment:
Maybe specify that the message is incomplete rather than no message is there?

Reply:
Ah, sorry my bad! I somehow completely mixed this up and thought to read the description of a status register here. But this is of course the control register where software defines that no message will follow but only AD etc. This makes sense of course, sorry!

Reply:
No problem.